### PR TITLE
[python] minor improvements

### PR DESCRIPTION
**Pull request by changhiskhan**
1. Image.create supports bytearray input as well
2. LanceDataset leaves str unconverted in record mode (filter column)

### CODE REVIEW

File: python/lance/pytorch/data.py [hunk 1]
```diff
@@ -43,6 +43,8 @@ def _data_to_tensor(data: Any) -> Union[torch.Tensor, PIL.Image.Image]:
         return data.to_pil()
     elif isinstance(data, dict):
         return {k: to_tensor(v) for k, v in data.items()}
+    elif isinstance(data, str):
```
Comment (eddyxu) [inline python/lance/pytorch/data.py:46]: What is the use case for this? Pytorch model can not take in strings, no valid `torch.Tensor` type for string.

File: python/lance/tests/test_types.py [hunk 2]
```diff
@@ -218,3 +218,10 @@ def test_pickle(tmp_path):
         pickle.dump(img, fh)
     with (tmp_path / "image").open("rb") as fh:
         assert img == pickle.load(fh)
+
+    img = Image.create(bytearray(b"bytes"))
```
Comment (eddyxu) [inline python/lance/tests/test_types.py:222]: is the change to support both bytes and bytearray?

File: python/lance/pytorch/data.py (hunk 1 above)
Comment (changhiskhan) [reply python/lance/pytorch/data.py:46]: so say you have `LanceDataset(uri, filter=pc.field("split") == "train")`

then the split column is included in the dataset and in record mode we get "train" as the value.

File: python/lance/tests/test_types.py (hunk 2 above)
Comment (changhiskhan) [reply python/lance/tests/test_types.py:222]: yeah, previously we only supported bytes but duckdb reads it out as bytearray so this is just a minor convenience

**Review by eddyxu — APPROVED** [review]

**Pull request merged**